src/control/ioserver/config.go

```diff
@@ -0,0 +1,740 @@
+package ioserver
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+type (
+	// ValidationError represents an error found while validating
+	// an option
+	ValidationError struct {
```

Comment:
do not want to store an actual error type rather than just a message? then you could have other context such as a stack trace

Reply:
Possibly. I don't think it would take anything away to make it an error type. My original intent was for ValidationErrors to be specifically about malformed option values, but now that custom validators may do things via os.Stat() and the like it might make more sense, yeah.